src/lib/learn-client/api/bookmark/get-bookmarks.ts

```diff
@@ -0,0 +1,49 @@
+import { get, toError } from 'lib/learn-client'
+import { SessionData } from 'types/auth'
+import { BOOKMARK_API_ROUTE } from '.'
+import { ApiBookmark, ApiTutorial } from '../api-types'
+
+interface GetBookmarksOptions {
+	accessToken: SessionData['accessToken']
+	tutorialIds?: ApiTutorial['id'][]
+}
+
+type GetBookmarksResult = ApiBookmark[]
+
+/**
+ * Fetches bookmarks based on given `tutorialIds`. If none are provided, then
+ * all bookmarks will be fetched. If some `tutorialIds` are provided, then they
+ * will be passed in the comma-separated `tutorialIds` query parameter.
+ *
+ * If successful, then an array of bookmark objects is returned. Otherwise
+ * throws an error.
+ *
+ * https://digital-api-specs.vercel.app/learn#tag/Bookmarks/paths/~1bookmarks/get
+ */
+const getBookmarks = async ({
```

Comment:
💭 I wonder if it would be nice to split this into two distinct functions, `getBookmarks` where the `tutorialIds` array is required. And then `getAllBookmarks`. To me, that would make it more clear in the codebase when we are calling for a specific set of bookmarks versus all of them. Thoughts?

Reply:
Still thinking on this one! I had originally decided to only create one function since there's only one endpoint, to make a 1:1 to mapping.

Reply:
Updated in https://github.com/hashicorp/dev-portal/pull/828/commits/1baea46b14953ddcadf1d06f0bb22bc2a7db69a6! There are now two methods instead of one:

- `getAllBookmarks`
- `getBookmarksByTutorialIds`